src/user/dto/learningHistory.dto.ts

```diff
@@ -1,31 +1,48 @@
-import { IsDefined, IsNotEmpty } from 'class-validator';
+import { IsNotEmpty } from "class-validator";
+import {
+    ConsumedUnitData,
+    LearningProgress,
+    UserProfile,
+    LearningProfile,
+    LearningHistory,
+    // PersonalizedLearningPath,
+} from "@prisma/client";
 
-import { UserDto } from './user.dto';
-import { LearningHistory } from '@prisma/client';
-import { LearningHistoryCreationDto } from './learningHistory-creation.dto';
+import { LearningHistoryCreationDto } from "./learningHistory-creation.dto";
 
 export class LearningHistoryDto extends LearningHistoryCreationDto {
-  @IsNotEmpty()
-  id!: string;
-  @IsDefined()
-  userId :string;
+    @IsNotEmpty()
+    id: string;
 
-  // workedAt   CareerProfile[] @relation("workedAt")
-  // workingNow CareerProfile[] @relation("workingNow")
+    constructor(
+        id: string,
+        startedLearningUnits: ConsumedUnitData[] | undefined,
+        learnedSkills: LearningProgress[] | undefined,
+        user: UserProfile | undefined,
+        userId: string,
+        learningProfile: LearningProfile[] | undefined,
+        // personalPaths: PersonalizedLearningPath[] | undefined,
+    ) {
+        super();
 
-  constructor(
-    id: string,
-    userId: string,
-    
-  ) {
-    super(userId);
-    this.id = id;
-  }
+        this.id = id;
+        this.startedLearningUnits = startedLearningUnits ?? undefined;
+        this.learnedSkills = learnedSkills ?? undefined;
+        this.user = user ?? undefined;
+        this.userId = userId;
+        this.learningProfile = learningProfile ?? undefined;
+        // this.personalPaths = PersonalizedLearningPath ?? undefined;
+    }
 
-  static createFromDao(lh: LearningHistory): LearningHistoryCreationDto {
-    return new LearningHistoryDto(
-      lh.id,
-      lh.userId
-    );
-  }
+    static createFromDao(lh: LearningHistory): LearningHistoryCreationDto {
+        return new LearningHistoryDto(
+            lh.id,
+            lh.startedLearningUnits,
+            lh.learnedSkills,
+            lh.user,
+            lh.userId,
+            lh.learningProfile,
+            //  lh.personalPaths,
+        );
+    }
```

Comment:
`startedLearningUnits`, `learnedSkills`, `user` & `learningProfile` do not exist on `LearningHistory`. Probably these are referenced fields of other tables (which may be selected via `include` statements. In this case, an intersection type is needed in Line 37, which specifies that these optiona files have been selected.

This is an example how to use an intersection type to specify that `nestedSkills` must be selected and `parentSkills` may be selected when passing a `Skill` object from `Prisma` to create a DTO:
```TypeScript
static createFromDao(
        skill: Skill & {
            nestedSkills: { id: string }[];
            parentSkills?: { id: string }[];
        },
    ): SkillDto {
        const dto: SkillDto = {
            id: skill.id,
            nestedSkills: skill.nestedSkills.map((element) => element.id),
            parentSkills: skill.parentSkills?.map((element) => element.id) || [],
            ...
        };
        return dto;
    }
```